docs/source/index.rst

```diff
@@ -701,13 +701,16 @@ When a function which has been declared to update those channels is called, the
             return "ok"
 
 
-.. function:: notify(channels)
+.. function:: notify(channels, delay=False)
```

Comment:
Here's what's going on with the new ``delay`` stuff.

People already familiar with Sideboard's pub/sub utilities will probably recognize the ``@subscribes`` and ``@notifies`` decorators.  This here is the ``notify()`` function which allows us to manually broadcast to a channel, which we sometimes use.

Consider the following problem: we're in the middle of a database transaction and want to trigger a notify after the transaction is complete.  We don't want to just call ``notify()``, because the background thread that handles broadcasts might run before we've committed our changes to the database.

When we originally considered this use case, we "solved" it by adding an optional ``delay`` argument to the ``notify`` method.  For example, plugins could say ``notify('my_channel', delay=1)`` to cause the notification to be triggered after 1 second, with the assumption that they'd pick an amount of time long enough to let the transaction finish.

This is kind of a terrible solution for several reasons:
- What if you pick an amount of time that's not long enough?  Maybe your transaction takes longer than you thought to complete.
- If you do pick a larger amount of time to solve that problem, it makes your notifications less responsive.
- In order to support this feature, our ``sideboard.lib.Caller`` class spins up an extra thread for every ``Caller`` instance, even though most callers will never need this feature.  (Admittedly, having a few additional threads sitting around doing nothing isn't much of a drain on system resources, but even still it adds a bunch of threads to the running system which makes watching ``top`` more annoying.)

I checked every Sideboard plugin we've ever written and we've literally never used this feature anyway!  However, we have recently started wanting to be able to do this ("notify at the end of the current RPC request"), so I figured this was a good time to bite the bullet and make this minor refactor.

Now we can say ``notify('my_channel', delay=True)`` which will cause the notify to occur after the current RPC method has completed.  This is technically a backwards-incompatible change, but since literally no one has ever used this feature that doesn't matter.  Also, this has been implemented in such a way that even if some third party was setting ``delay`` to an integer like in the old approach, their code would still work as they expected!